### PR TITLE
add user to firebase db on start command

### PR DESCRIPTION
**Pull request by Proladge**

### CODE REVIEW

File: server/src/bots/telegram/botResponse/startResponse.ts [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+import { getFullMenuKeyboard } from '../services/keyboard';
+import { greetUser } from '../../../messages/userMessage';
+import { showHelpInfoResponse } from './helpResponse';
+import { CallBackQueryHandler } from '../models';
+import { getUser, addUser } from '../../../services/domain/user'
+import { logger } from '../../../utils/logger';
+import {User} from '../../../models/user.model'
 
 export const startResponse: CallBackQueryHandler = async (bot, message, chatId) => {
+    let user = await getUser(chatId);
```
Comment (danbilokha) [inline server/src/bots/telegram/botResponse/startResponse.ts:10]: - type `let user: User = await getUser(chatId);`
- let's not force a user to wait for that. You can give a user the response and, meanwhile, write him in Firebase

File: server/src/bots/telegram/botResponse/startResponse.ts [hunk 2]
```diff
@@ -1,9 +1,33 @@
...
+        user = {
+            chatId,
+            userName: message.chat.username || '',
+            firstName: message.chat.first_name || '',
+            lastName: message.chat.last_name || '',
+            startedOn: Date.now()
+        };
+        try {
+            await addUser(user);
+            logger.log('info', `New user ${user.chatId} was successfully added`)
```
Comment (danbilokha) [inline server/src/bots/telegram/botResponse/startResponse.ts:21]: Use `getInfoMessage` fn in `getInfoMessage.ts` util's file

File: server/src/models/user.model.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export type User = {
```
Comment (danbilokha) [inline server/src/models/user.model.ts:1]: I am adding Telegram type's, let's wait for a while

File: server/src/services/domain/user.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import {getTelegramUser, addTelegramUser} from '../../../src/bots/telegram/services/storage'
+import { User } from '../../models/user.model';
+
+export const getUser = async (chatId: number): Promise<User> => (await getTelegramUser(chatId) ?? {});
+export const addUser = async (user: User): Promise<Array<User>> => (await addTelegramUser(user) ?? {});
```
Comment (danbilokha) [inline server/src/services/domain/user.ts:5]: I am not sure why you add domain wrapper here. My first hesitation would be that it's coupled with Telegram

File: server/src/bots/telegram/botResponse/startResponse.ts (hunk 1 above)
Comment (Proladge) [reply server/src/bots/telegram/botResponse/startResponse.ts:16]: 2. fixed

File: server/src/bots/telegram/botResponse/startResponse.ts (hunk 2 above)
Comment (Proladge) [reply server/src/bots/telegram/botResponse/startResponse.ts]: Logging approach reset 💥

File: server/src/models/user.model.ts (hunk 3 above)
Comment (Proladge) [reply server/src/models/user.model.ts:1]: Good

File: server/src/models/constants.ts [hunk 5]
```diff
@@ -51,3 +51,19 @@ export enum Status {
...
+//   Crit = 'crit', 
+  Error = 'error', 
+  Warning = 'warn', 
+//   Notice = 'notice', 
+  Info = 'info', 
+  Debug = 'debug', //????????
+//   Trace = 'trace'
+}
+
+export enum LogCategory {
```
Comment (danbilokha) [inline server/src/models/constants.ts:67]: empty

File: server/src/models/constants.ts [hunk 6]
```diff
@@ -51,3 +51,19 @@ export enum Status {
     Deaths = 'deaths',
     Recovered = 'recovered',
 }
+
+export enum LogLevel {
+//   Emerg = 'emerg',
```
Comment (danbilokha) [inline server/src/models/constants.ts:56]: what are these?

File: server/src/services/domain/user.ts (hunk 4 above)
Comment (danbilokha) [reply server/src/services/domain/user.ts:5]: Still is not resolved

**Comment by danbilokha** [comment]
cool! 👍 Some comments and your PR doesn't pass some checks, take a look at them as well

File: server/src/models/constants.ts (hunk 6 above)
Comment (Proladge) [reply server/src/models/constants.ts:56]: These are log levels supported by winston. But when I try to log messages to loggly. Just those uncommented works. I'm still in process of figuring out how to fix it

File: server/src/models/constants.ts (hunk 5 above)
Comment (Proladge) [reply server/src/models/constants.ts]: fixed

**Pull request merged**